### DISCUSSION TITLE
Seeking help regarding Lifecycle functions are not getting called.

### DISCUSSION THREAD

**Discussion by gulabsagevadiya** [arkivanov/Decompose#631] [Q&A]
## MyCode Declaration with dependancy.  -- Issue shown at bottom

`extensionsComposeJetbrains = "2.2.2-compose-experimental"`
`decompose= "2.2.2"`

`commonMain.dependencies {`
`   //Other deps`
`   implementation(libs.decompose)`
`   implementation(libs.decompose.extension.compose)`
`}`

IOS Main

`fun MainViewController() = ComposeUIViewController {`
`    val root = remember {`
`        AppComponent(DefaultComponentContext(lifecycle = LifecycleRegistry()))`
`    }`
`    App(root)`
`}`

CommonMain

`@Composable`
`fun App(rootComponent: AppComponent) {`
`    AppTheme {`
`        val childStack by rootComponent.childStack.subscribeAsState()`
`        Children(`
`            stack = childStack,`
`            animation = stackAnimation(fade())`
`        ) { child ->`
`            when (val instance = child.instance) {`
`                is AppComponent.Child.SplashScreenChild -> SplashScreenView(instance.component)`
`                is AppComponent.Child.MainScreenChild -> MainScreenView(instance.component)`
`            }`
`        }`
`    }`
`}`

## App Component Declaration

`import com.arkivanov.decompose.ComponentContext`
`import com.arkivanov.decompose.router.stack.StackNavigation`
`import com.arkivanov.decompose.router.stack.childStack`
`import com.arkivanov.essenty.lifecycle.doOnResume`
`import kotlinx.serialization.Serializable`

`class AppComponent(`
`    componentContext: ComponentContext`
`) : ComponentContext by componentContext {`

`    private val appNavigation = StackNavigation<Configuration>()`

`    val childStack = childStack(`
`        source = appNavigation,`
`        serializer = Configuration.serializer(),`
`        initialConfiguration = Configuration.SplashScreenConfig,`
`        handleBackButton = true,`
`        childFactory = ::createChild`
`    )`

`    private fun createChild(`
`        config: Configuration,`
`        context: ComponentContext`
`    ): Child {`
`        return when (config) {`
`            Configuration.SplashScreenConfig -> Child.SplashScreenChild(SplashComponent(context))`
`            Configuration.MainScreenConfig -> Child.MainScreenChild(MainComponent(context))`
`        }`
`    }`

`    sealed class Child {`
`        data class SplashScreenChild(val component: SplashComponent) : Child()`
`        data class MainScreenChild(val component: MainComponent) : Child()`
`    }`

`    @Serializable`
`    sealed class Configuration {`

`        @Serializable`
`        data object SplashScreenConfig : Configuration()`

`        @Serializable`
`        data object MainScreenConfig : Configuration()`

`    }`
`}`

## Lifecycle event not working

`class MainComponent(componentContext: ComponentContext): ComponentContext by componentContext {`
`    init {`
`        lifecycle.doOnResume {`
`            print("OnResume Called") // This is not getting printed`
`        }`
`        print("Init Called") //this is gets printed`
`    }`
`}`

**Comment by arkivanov** [comment]
This is because you are just passing `LifecycleRegistry`, and it never gets called. Please see the [docs](https://arkivanov.github.io/Decompose/getting-started/quick-start/#ios-with-swiftui-with-the-experimental-applicationlifecycle) how to setup the root component properly.
